backend/dataall/modules/notebooks/aws/client.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from dataall.aws.handlers.sts import SessionHelper
+from dataall.modules.notebooks.db.models import SagemakerNotebook
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger(__name__)
+
+
+class SagemakerClient:
+    """
+    A Sagemaker notebooks proxy client that is used to send requests to AWS
+    """
+
+    def __init__(self, notebook: SagemakerNotebook):
+        session = SessionHelper.remote_session(notebook.AWSAccountId)
+        self._client = session.client('sagemaker', region_name=notebook.region)
+        self._instance_name = notebook.NotebookInstanceName
+
+    def get_notebook_instance_status(self) -> str:
+        """Remote call to AWS to check the notebook's status"""
+        try:
+            response = self._client.describe_notebook_instance(
+                NotebookInstanceName=self._instance_name
+            )
+            return response.get('NotebookInstanceStatus', 'NOT FOUND')
+        except ClientError as e:
+            logger.error(
+                f'Could not retrieve instance {self._instance_name} status due to: {e} '
+            )
+            return 'NOT FOUND'
+
+    def presigned_url(self):
+        """Creates a presigned url for a notebook instance by sending request to AWS"""
+        try:
+            response = self._client.create_presigned_notebook_instance_url(
+                NotebookInstanceName=self._instance_name
+            )
+            return response['AuthorizedUrl']
+        except ClientError as e:
+            raise e
+
+    def start_instance(self):
+        """Starts the notebooks instance by sending a request to AWS"""
+        try:
+            status = self.get_notebook_instance_status()
+            self._client.start_notebook_instance(NotebookInstanceName=self._instance_name)
+            return status
+        except ClientError as e:
+            return e
+
+    def stop_instance(self) -> None:
+        """Stops the notebooks instance by sending a request to AWS"""
+        try:
+            self._client.stop_notebook_instance(NotebookInstanceName=self._instance_name)
+        except ClientError as e:
+            raise e
+
+
+def client(notebook: SagemakerNotebook) -> SagemakerClient:
+    """Factory method to retrieve the client to send request to AWS"""
+    return SagemakerClient(notebook)
```

Comment:
I would rename the client to 'sagemaker_client' for cases that use more than one AWS service
Also, do you think that we could have a`aws` package as part of the `common` package? Some of the aws clients are used in several modules. For example the IAM handler

Reply:
There is also the option of having the whole `aws` package as part of `common` and remove it from the modules, but as downside we will end up with "module code" outside of the module directory

Reply:
Tbh, I was thinking about naming it as `sagemaker_client`. The only think that stopped me is that we already have a convention to name  submodules like `resolvers`, `queries`, and etc. 
I think we should be consistent in the naming of all submodules. If we are going to rename `client` to `sagemaker_client` we should rename `mutations` to `sagemaker_mutations`.  If we don't have such convention we end up having some file with prefixes and others not.

I see a huge plus from your suggestion: it should help easier navigate across the project (when you want to open a specific file), but as a downside, we will have more changes in the PR 

How would you like to proceed on this one? 🙂 


About `aws` module. I don't see an advantage of having it in `common`. I think every module will have their own `client.py` (or `module_client.py` as we decide)  in `aws` submodule instead of having a bigger module of all clients in `common`. It will help us encapsulate the logic of wrapper upon boto3 for each specific module. 
However, if some modules will require to use the same code, we can extract this code to `common`. 
Let me know what you think 🙂